assets/model_monitoring/components/tests/unit/test_compute_data_quality_metrics.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""This file contains unit tests for the Model Monitor Data Quality Compute Metric component."""
+
+import pytest
+from pyspark.sql.functions import current_timestamp
+from pyspark.sql.types import Row
+from src.data_quality_compute_metrics.compute_data_quality_metrics import (
+    compute_dtype_violation_count_modify_dataset,
+    modify_dataType)
+from tests.e2e.utils.io_utils import create_pyspark_dataframe
+
+
+df_without_violation = [
+        ( True, 41.2, 101001, "2021-03-11", 2.345, \
+          345.66, 234, 45634, 3877, "string", \
+          "a", "a")
+        ]
+columns = ["feature_boolean", "feature_double", "feature_binary", "feature_date", "feature_decimal", \
+           "feature_float", "feature_integer", "feature_long", "feature_short", "feature_string", \
+           "feature_char", "feature_varchar"]
+df = create_pyspark_dataframe(df_without_violation, columns)
+df_without_violation = df.withColumn("feature_timestamp", current_timestamp())
+
+data_with_violation = [
+        ( "*", "*", "*", "*", \
+          "*", "*", "*", "*", None, \
+          None, None ),
+        ( "*", "*", "*", "*", \
+          "*", "*", "*", "*", 1, \
+          1, 1 )
+        ]
+columns = ["feature_boolean", "feature_double",  "feature_date", "feature_decimal", \
+           "feature_float", "feature_integer", "feature_long", "feature_short", "feature_string", \
+           "feature_char", "feature_varchar"]
+df_with_violation = create_pyspark_dataframe(data_with_violation, columns)
+
+columns = ['featureName',  'dataType']
+data_stats = [
+            ('feature_timestamp', 'timestamp'),
+            ('feature_boolean', 'boolean'),
+            ('feature_double', 'double'),
+            ('feature_binary', 'binary'),
+            ('feature_date', 'date'),
+            ('feature_decimal', 'decimal'),
+            ('feature_float', 'float'),
+            ('feature_integer', 'integer'),
+            ('feature_long', 'long'),
+            ('feature_short', 'short'),
+            ('feature_string', 'string'),
+            ('feature_char', 'char(30)'),
```

Comment:
You tested all the cast be very successful, right?

Reply:
Yes, when I test the cast for "char" and "varchar". It showed that I have to case with char(int) and varchar(int), so I added an int in char and varchar